server/src/main/java/org/opensearch/action/admin/cluster/remotestore/stats/RemoteStoreStatsResponse.java

```diff
@@ -41,30 +50,85 @@ public RemoteStoreStatsResponse(
         List<DefaultShardOperationFailedException> shardFailures
     ) {
         super(totalShards, successfulShards, failedShards, shardFailures);
-        this.shards = shards;
+        this.remoteStoreStats = shards;
     }
 
-    public RemoteStoreStats[] getShards() {
-        return this.shards;
+    public RemoteStoreStats[] getRemoteStoreStats() {
+        return this.remoteStoreStats;
     }
 
     public RemoteStoreStats getAt(int position) {
-        return shards[position];
+        return remoteStoreStats[position];
+    }
+
+    public Map<String, Map<Integer, List<RemoteStoreStats>>> groupByIndexAndShards() {
+        if (indexWiseStats != null) {
+            return indexWiseStats;
+        }
```

Comment:
In what cases would `indexWiseStats != null` be true?

Reply:
Removing this `indexWiseStats` var altogether in-lieu of the new optimized `Map.computeIfAbsent` based logic